checkov/common/bridgecrew/platform_integration.py

```diff
@@ -180,7 +182,12 @@ def persist_repository(self, root_dir):
                 if file_extension in SUPPORTED_FILE_EXTENSIONS:
                     full_file_path = os.path.join(root_path, file_path)
                     relative_file_path = os.path.relpath(full_file_path, root_dir)
-                    self._persist_file(full_file_path, relative_file_path)
+                    if not files:
+                        # upload every file since -f was not used.
+                        self._persist_file(full_file_path, relative_file_path)
+                    elif full_file_path in files:
```

Comment:
`files` is a list of absolute file paths, so can't we just iterate it directly if it is not null?

Reply:
Changed it so that it doesn't walk through all the nested directories. 
However, the relative path logic is still intact. If you run -
`checkov -f ../goat/terragoat/terraform/aws/s3.tf -f ~/dev/goat/terragoat/terraform/azure/sql.tf --bc-api-key <hush-hush> --repo-id try_bridgecrew/terragoat
`
It will result in the following folder structure:
── terraform
    └── aws
       └── s3.tf
    └──  azure
       └── sql.tf